docs/proposals/naming.md

```diff
@@ -0,0 +1,44 @@
+# Enhancement Title
+
+> [Issue #441](https://github.com/redhat-et/issues/441)
+
+## Summary
+
+Proposal for using consistent naming for the major components of the Apex stack
+
+## Proposal
+
+Currently we are are using multiple names for different components across our documentation and code, such as
+
+* Apex Stack is referred as Apex Service, Apex ApiServer, Apex Controller, Apex Control Plane, Apex SaaS
+* Apex Agent is referred as hub-router, relay, agent, discovery node
+
+It would be really helpful to write a coherent documentation and code if we can zero down on the naming of these components.
+
+Here is my initial proposal to start the discussion
+
+* Apex Service: It's an umbrella term to refer to the entire stack, that contains following individual components (Named as is).
+  * Apex ApiServer
+  * Apex Frontend
+  * Apex Database
+  * Apex Ipam
+  * Apex ApiProxy
+  * Apex backend-cli
+  * Apex backend-web
+* Apex Agent: Agent running on each node, just doing Agent's job.
+  * Binary Naming: We need to go one more level down to clarify the binary naming as we are planning to separate the relay functionality out of agent.
+    * apexd: Agent binary (running as a daemon) that does the magic of connectivity.
+    * apex: A command line utility to do read/write operation through Apexd
+* Apex ICE: A node running the discovery and relay function.
+  * Binary Naming:
+    * apexiced: Binary (running as a daemon) that does the magic of discovery and relay.
+    * apexice: A command line utility to do read/write operation through Apexiced
```

Comment:
ICE is the protocol the client uses to discover, prioritize, and test the endpoint candidate addresses of peers.  I don't really like it to describe a node that would be running stun and relay functionality.  I would rather call it something like relay.

Reply:
yeah makes sense. Relay was an option i was also thinking, but again it's kind of restrict to one functionality, and as of now it's doing more than that. Let's continue discussion on #440